### PR TITLE
pygobject: 3.24.1 → 3.26.1

### PR DESCRIPTION
**Pull request by jtojnar**
###### Motivation for this change
Just an update, trying if it does not fix syncthing-gtk freeze. It does not.

* https://github.com/GNOME/pygobject/blob/pygobject-3-26/NEWS
* https://pycairo.readthedocs.io/en/latest/changelog.html#v1-15-4

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by jtojnar** [comment]
The thing I do not like about this is the `postPatch` section in the pycairo expression. Is there a better way to set the [install_base](https://github.com/pygobject/pycairo/blob/master/setup.py#L152)? How do other python packages solve this?

File: pkgs/development/python-modules/pycairo/default.nix [hunk 1]
```diff
@@ -1,50 +1,27 @@
...
+    owner = "pygobject";
+    repo = "pycairo";
+    rev = "v${version}";
+    sha256 = "02vzmfxx8nl6dbwzc911wcj7hqspgqz6v9xmq6579vwfla0vaglv";
+  };
+
+  postPatch = ''
+    # prefix is ${python} by default for some reason
+    substituteInPlace setup.py --replace '"prefix": self.install_base' "'prefix': '$out'"
+  '';
```
Comment (FRidh) [inline pkgs/development/python-modules/pycairo/default.nix]: maybe the following works
```
setupPyBuildFlags = [ "--install-base=$out" ];
```

Comment (FRidh) [reply pkgs/development/python-modules/pycairo/default.nix]: https://stackoverflow.com/questions/5351086/what-exactly-does-the-install-base-parameter-to-setup-py-install-mean

Comment (jtojnar) [reply pkgs/development/python-modules/pycairo/default.nix]: Does not seem to work – `no such option: --install-base`. I also tried

    installFlags = [ "--install-base=$out" ];

with the same result, and

    installFlags = [ "--install-option=\"--install-base=$out\"" ];

which at least builds but the `${python}` path is still used.

Comment (jtojnar) [reply pkgs/development/python-modules/pycairo/default.nix]: The option seems to be distutils-specific, while this package seems to be using setuptools.

Comment (jtojnar) [reply pkgs/development/python-modules/pycairo/default.nix]: Apparently, the problem is caused by pycairo reading an install flag during the build of the wheel. Just like with [passing build flags to bdist_wheel](https://github.com/NixOS/nixpkgs/blob/c0c503ca685f5a70390c8f2bb850e8b9690fab17/pkgs/development/interpreters/python/build-python-package-setuptools.nix#L9), setuptools do not allow passing the install flags either – other than by using hacks like `python setup.py install --root=_temp --prefix=/foo bdist_wheel` ([source](https://quodlibet.duckdns.org/irc/pygobject/python_2017-11-27.log.html#t18:55:09)). Let’s keep patching the source for now.

**Comment by jtojnar** [comment]
Verified by running `d-feet`.

**Pull request merged**